crates/amaru-ledger/src/state/transaction.rs

```diff
@@ -85,197 +94,213 @@ pub fn apply(
     span.exit();
 }
 
-/// A trait to extract inputs, outputs and fees from a transaction; based on whether it is a failed
-/// transaction or not. It is meant to provide a unified interface that makes the parallel between
-/// the two cases.
-trait InputsOutputs<const IS_FAILED: bool> {
```

Comment:
This felt like unnecessary abstraction so I removed it, open to rollback

Reply:
Felt cute at the time; completely unnecessary I concede. Two functions is more boring. We like boring.